lib/Recur/RRuleIterator.php

```diff
@@ -588,6 +588,11 @@ protected function nextYearly() {
                     (int)$currentDayOfMonth
                 );
 
+                // maximum year 10000 should be safe to stop (2016 + max 3500 yearly occurrences)
+                if ((int)$currentYear > 10000) {
+                    throw new NoInstancesException('This recurrence rule does not generate any valid instances, stopped in year 10000');
```

Comment:
I'm also not sure if an exception is really correct. The standard says that if a recurrence rule generates dates that are invalid (february 29th for some years for example) then those instances should just be skipped. So it's not necessarily "invalid" even if it's actually pretty wrong.

I think if we use the MAX_DATE setting we should simply just exit the loop and don't generate additional instances. That brings it in line with how the rest of the standard behaves I believe.